xcode/Subconscious/Shared/Models/Petname.swift

```diff
@@ -17,6 +17,7 @@ public struct Petname:
     LosslessStringConvertible
 {
     private static let petnameRegex = /([\w\d\-]+)(\.[\w\d\-]+)*/
+    private static let petnameSuffixRegex = /^(?<petname>(.*?))(?<separator>-+)?(?<suffix>(\d+))?$/
```

Comment:
Could we simplify this by saying the separator is always one dash?

Reply:
I played with this but if you look at the test cases this was the best way I could find to produce the "correct" petname in all cases.

The corner cases I'm was covering with this are things like `ben-----`  which should become `ben-----1` and `sarah99` which should become `sarah99-1` _not_ `sarah100` etc.

Reply:
👍  that make sense. To clarify, I meant it could be possible to simplify the regex to match a single dash, rather than all. E.g.

Current Regex (matches many dashes):

<img width="870" alt="Screen Shot 2023-04-18 at 8 55 05 AM" src="https://user-images.githubusercontent.com/58421/232784434-99bce6e7-bf88-4aec-a195-fa79bac9f19a.png">

Possible alternative (matches just the one dash related to suffix):

<img width="871" alt="Screen Shot 2023-04-18 at 8 54 51 AM" src="https://user-images.githubusercontent.com/58421/232784528-ed32612a-55f6-4e41-941e-fe2edd4778c6.png">

These are functionally equivalent for our purposes, but the second seemed closer to the intended logic to my eye.

Caveat: we are deep into nit territory, and no change or action is needed.